Mage.Sets/src/mage/cards/k/KardurDoomscourge.java

```diff
@@ -101,3 +92,57 @@ public boolean canAttack(Permanent attacker, UUID defenderId, Ability source, Ga
         return !defenderId.equals(source.getControllerId());
     }
 }
+
+class KardurDoomscourgeTriggeredAbility extends TriggeredAbilityImpl {
+
+    private List<UUID> attackers = null;
```

Comment:
Do not store changeable data in the ability objects. Abilities are copies all around the code. You must use GameState and source related ids to store data (setValue, getValue).